flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/TransformFilterProcessor.java

```diff
@@ -67,15 +73,22 @@ public static TransformFilterProcessor of(
             TransformFilter transformFilter,
             String timezone,
             List<UserDefinedFunctionDescriptor> udfDescriptors,
-            List<Object> udfFunctionInstances) {
+            List<Object> udfFunctionInstances,
+            SupportedMetadataColumn[] supportedMetadataColumns) {
         return new TransformFilterProcessor(
-                tableInfo, transformFilter, timezone, udfDescriptors, udfFunctionInstances);
+                tableInfo,
+                transformFilter,
+                timezone,
+                udfDescriptors,
+                udfFunctionInstances,
+                supportedMetadataColumns);
```

Comment:
Feels like we need something like `TransformContext` to gather all these things together instead of extending arguments lists forever, but it's not a must in this PR.

Reply:
Thanks for this suggestion. Let open a new issue for this optimization.